src/rust/plugin-work-queue/src/server.rs

```diff
@@ -24,111 +31,266 @@ use rust_proto::plugin_work_queue::{
+        let data = request.execution_job.data;
+
+        self.queue
+            .put_generator_message(plugin_id, data, tenant_id)
+            .await?;
+
+        Ok(PutExecuteGeneratorResponse {})
     }
 
     #[allow(dead_code)]
     async fn put_execute_analyzer(
         &self,
-        _request: PutExecuteAnalyzerRequest,
+        request: PutExecuteAnalyzerRequest,
     ) -> Result<PutExecuteAnalyzerResponse, PluginWorkQueueError> {
-        todo!()
+        let tenant_id = request.execution_job.tenant_id;
+        let plugin_id = request.execution_job.plugin_id;
+        let data = request.execution_job.data;
+
+        self.queue
+            .put_analyzer_message(plugin_id, data, tenant_id)
+            .await?;
+
+        Ok(PutExecuteAnalyzerResponse {})
     }
 
     #[allow(dead_code)]
     async fn get_execute_generator(
         &self,
         _request: GetExecuteGeneratorRequest,
     ) -> Result<GetExecuteGeneratorResponse, PluginWorkQueueError> {
-        todo!()
+        let message = self.queue.get_analyzer_message().await?;
+        let message = match message {
+            Some(message) => message,
+            None => {
+                return Ok(GetExecuteGeneratorResponse {
+                    execution_job: None,
+                    request_id: 0,
+                })
+            }
+        };
+        let execution_job = ExecutionJob {
+            tenant_id: message.request.tenant_id,
+            plugin_id: message.request.plugin_id,
+            data: message.request.pipeline_message,
+        };
+        Ok(GetExecuteGeneratorResponse {
+            execution_job: Some(execution_job),
+            request_id: message.request.execution_key.into(),
+        })
     }
 
     #[allow(dead_code)]
     async fn get_execute_analyzer(
         &self,
         _request: GetExecuteAnalyzerRequest,
     ) -> Result<GetExecuteAnalyzerResponse, PluginWorkQueueError> {
-        todo!()
+        let message = self.queue.get_analyzer_message().await?;
+        let message = match message {
+            Some(message) => message,
+            None => {
+                return Ok(GetExecuteAnalyzerResponse {
+                    execution_job: None,
+                    request_id: 0,
+                })
+            }
+        };
+        let execution_job = ExecutionJob {
+            tenant_id: message.request.tenant_id,
+            plugin_id: message.request.plugin_id,
+            data: message.request.pipeline_message,
+        };
+        Ok(GetExecuteAnalyzerResponse {
+            execution_job: Some(execution_job),
+            request_id: message.request.execution_key.into(),
+        })
+    }
+
+    #[allow(dead_code)]
+    async fn acknowledge_generator(
+        &self,
+        request: AcknowledgeGeneratorRequest,
+    ) -> Result<AcknowledgeGeneratorResponse, PluginWorkQueueError> {
+        self.queue
+            .ack_generator_failure(request.request_id.into())
```

Comment:
Not immediately obvious to me why this should be an `ack_*_failure`

Reply:
It shouldn't be. I realize there's an issue with the Ack concept currently.